controller/volume_attachment_controller_test.go

```diff
@@ -1,400 +1,400 @@
+	tc.vol.Status.State = longhorn.VolumeStateAttached
+	tc.copyCurrentToExpect()
+	tc.expectedVolAttachment.Status.AttachmentTicketStatuses = map[string]*longhorn.AttachmentTicketStatus{}
+	tc.expectedVol.Spec.NodeID = ""
+	testCases["test case 6: detach: basic"] = tc
+	///////////////////////////////////////////////////////////////////
+
+	///////////////////////////////////////////////////////////////////
+	tc = generateVolumeAttachmentTestCaseTemplate(TestVolumeName)
+	tc.volAttachment.Spec.AttachmentTickets = map[string]*longhorn.AttachmentTicket{
+		"attachment-01": &longhorn.AttachmentTicket{
+			ID:         "attachment-01",
+			Type:       longhorn.AttacherTypeCSIAttacher,
+			NodeID:     TestNode1,
+			Parameters: map[string]string{},
+			Generation: 0,
+		},
+	}
+	tc.volAttachment.Status.AttachmentTicketStatuses = map[string]*longhorn.AttachmentTicketStatus{
+		"attachment-01": &longhorn.AttachmentTicketStatus{
+			ID:        "attachment-01",
+			Satisfied: true,
+			Conditions: types.SetConditionWithoutTimestamp([]longhorn.Condition{},
+				longhorn.AttachmentStatusConditionTypeSatisfied, longhorn.ConditionStatusTrue, "", ""),
+			Generation: 0,
+		},
+		"attachment-02": &longhorn.AttachmentTicketStatus{
+			ID:        "attachment-02",
+			Satisfied: true,
+			Conditions: types.SetConditionWithoutTimestamp([]longhorn.Condition{},
+				longhorn.AttachmentStatusConditionTypeSatisfied, longhorn.ConditionStatusTrue, "", ""),
+			Generation: 0,
+		},
+	}
+	tc.vol.Status.OwnerID = TestNode1
+	tc.vol.Spec.NodeID = TestNode1
+	tc.vol.Status.CurrentNodeID = TestNode1
+	tc.vol.Status.State = longhorn.VolumeStateAttached
+	tc.copyCurrentToExpect()
+	delete(tc.expectedVolAttachment.Status.AttachmentTicketStatuses, "attachment-02")
+	testCases["test case 7: detach: detach while there are still other attachments requesting the same node"] = tc
+	///////////////////////////////////////////////////////////////////
+
+	///////////////////////////////////////////////////////////////////
+	tc = generateVolumeAttachmentTestCaseTemplate(TestVolumeName)
+	tc.volAttachment.Spec.AttachmentTickets = map[string]*longhorn.AttachmentTicket{
+		"attachment-01": &longhorn.AttachmentTicket{
+			ID:         "attachment-01",
+			Type:       longhorn.AttacherTypeCSIAttacher,
+			NodeID:     TestNode1,
+			Parameters: map[string]string{},
+			Generation: 0,
+		},
+	}
+	tc.vol.Status.OwnerID = TestNode1
+	tc.vol.Spec.NodeID = TestNode1
+	tc.vol.Spec.DisableFrontend = true
+	tc.vol.Status.CurrentNodeID = TestNode1
+	tc.vol.Status.State = longhorn.VolumeStateAttached
+	tc.copyCurrentToExpect()
+	tc.expectedVolAttachment.Status.AttachmentTicketStatuses = map[string]*longhorn.AttachmentTicketStatus{
+		"attachment-01": &longhorn.AttachmentTicketStatus{
+			ID:        "attachment-01",
+			Satisfied: true,
+			Conditions: types.SetConditionWithoutTimestamp([]longhorn.Condition{},
+				longhorn.AttachmentStatusConditionTypeSatisfied, longhorn.ConditionStatusTrue, "", ""),
```

Comment:
Why is this true? Isn't the volume still state detaching?

Reply:
The volume start detaching because the volume.Spec.NodeID is just reset. However, the vol.Status is still attached at this moment so the AD ticket refect that fact.